src/main/kotlin/io/github/gradlenexus/publishplugin/internal/BasicActionRetrier.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin.internal
+
+import io.github.gradlenexus.publishplugin.RetryingConfig
+import io.github.gradlenexus.publishplugin.StagingRepository
+import net.jodah.failsafe.Failsafe
+import net.jodah.failsafe.RetryPolicy
+
+open class BasicActionRetrier<R>(retryingConfig: RetryingConfig, stopFunction: (R) -> Boolean) : ActionRetrier<R> {
+
+    private val retrier: RetryPolicy<R> = RetryPolicy<R>()
+            .handleResultIf(stopFunction)
+            .withMaxRetries(retryingConfig.maxRetries.get())
+            .withDelay(retryingConfig.delayBetween.get())
```

Comment:
Failsafe is nice and probably well tested, but I would like to test that I pass the parameters correctly. Functional tests check retrying in general (if it occurs), but I would like to test some corner cases (such as a reaction on particular exceptions) at the unit (testing) layer.
Unfortunately, `RetryingConfig` relies on the Gradle API (for `Property`). I initially modeled it as a implementation of a generic interface which just provides values, but it complicated the code. Do you have any better/simpler idea how would we make `BasicActionRetries` less Gradle dependent?

Reply:
I think I would pass in `retryingConfig.maxRetries.get()` and `retryingConfig.delayBetween.get()` to the constructor instead of `RetryingConfig`.

Reply:
That's definitely much simpler solution :-). Probably we will not have (much) more parameters passes here in the future.